src/alert.cpp

```diff
@@ -804,14 +821,15 @@ namespace libtorrent {
 		, int listen_port
 		, int op
 		, error_code const& ec
-		, socket_type_t t)
+		, libtorrent::socket_type_t t)
 		: error(ec)
 		, operation(op)
-		, sock_type(t)
+		, socket_type(t)
 		, address(listen_addr)
 		, port(listen_port)
 #ifndef TORRENT_NO_DEPRECATE
 		, endpoint(listen_addr, listen_port)
+		, sock_type(static_cast<socket_type_t>(sock_type_idx(t)))
```

Comment:
you could just say: ``sock_type(int(t))`` or, if that doesn't work: ``static_cast<int>(t)``